### ISSUE TITLE
[feature request] Count of different tag values

### ISSUE THREAD

**Issue by The-Nik** [influxdata/influxdb#3760]
Hi together,

with the query

    SHOW TAG VALUES WITH KEY = id 

I get a list of all different tag values of my database.

Now I need to count them. An example, there query above returns:

    name:idTagValues
    ----------------------- 
    id
    myId1
    myId2
    myId3

I need the result "3". Is their any way to get it? I tried some possibilities but everything fails:
An idea for further releases is to combine two queries, so the second one references at the first query's result:

    SHOW TAG VALUES WITH KEY = id; SELECT count(id) FROM idTagValues

Any idea to realize my request?

Thanks :-)

**Comment by beckettsean** [comment]
There's no way to do this currently.

**Comment by The-Nik** [comment]
Maybe an idea for further release? I would love to see that I can reference to a query result from an other query.

**Comment by beckettsean** [comment]
@The-Nik yes, I changed this to a feature request and left it open so we can discuss whether and when to implement this feature.

**Comment by rvrignaud** [comment]
This would be useful for us too +1

**Comment by larrywalker** [comment]
+1 for us also.  The only "workaround" is to store data as a tag and field.  Seems wasteful.

**Comment by wojciechszela** [comment]
+1 would help us a lot

**Comment by mouseratti** [comment]
I need this too

**Comment by sbengo** [comment]
+1

**Comment by kbespalov** [comment]
+1.

**Comment by wegel** [comment]
+1

**Comment by saboteurkid** [comment]
+1

**Comment by TechBK** [comment]
+1

**Comment by codepattern** [comment]
+1

**Comment by dstreppa** [comment]
+1

**Comment by lpalm** [comment]
+1

**Comment by vw-d3** [comment]
+1

**Comment by ngong** [comment]
+1

**Comment by supershal** [comment]
+1

**Comment by gargooza** [comment]
+1

**Comment by StephenPCG** [comment]
+1

**Comment by kittee** [comment]
+1

**Comment by cloudnull** [comment]
+1

**Comment by jsternberg** [comment]
We're going to implement this with #7195.

**Comment by kainoj** [comment]
It seems that update is on the way but I found this nested query:
```sql
SELECT count("count") FROM (SELECT count(temperature) FROM mydb GROUP BY sensor)
```
where `sensor` is a tag I want to count and `temperature` is float value associated with the tag.

It looks weird but it works :laughing:

**Comment by jasper-zhang** [comment]
#8636 `SHOW CARDINALITY` functionality is coming.

**Comment by mvadu** [comment]
@benbjohnson #8636 looks like merged (on Aug 16 itself) , but on latest stable release v1.3.7 [2017-10-26]  
`SHOW TAG VALUES CARDINALITY FROM "<measurement>" WITH KEY = '<tag key>'` or `SHOW TAG KEY CARDINALITY` is not working. e.g. later gives me error `error parsing query: found KEY, expected KEYS, VALUES at line 1, char 10`

**Comment by timhallinflux** [comment]
`SHOW CARDINALITY` is available starting with the v1.4 release.

**Comment by vlyalcin** [comment]
I know the issue has been closed, but I find it useful to share a link and example since this page still appears on google search. 

This solves the question: https://docs.influxdata.com/influxdb/v1.8/query_language/spec/#show-cardinality

Example in my case:
> SHOW TAG VALUES CARDINALITY WITH KEY = "Client_Name"
name: CUSTOMER_BACKUP
count
-----
6062
